doc/specs/stdlib_hash_maps.md

```diff
@@ -0,0 +1,3088 @@
+---
+title: Hash maps
+---
+
+# The `stdlib_32_bit_key_data_wrapper`, `stdlib_chaining_hash_map` and `stdlib_open_hash_map` modules
+
+(TOC)
+
+## Overview of hash maps
+
+The comparison of lexical entities or other objects for equality
+can be computationally expensive.
+This cost is often reduced by computing a near unique integer value,
+termed a hash code, from the structure of the object using a procedure
+termed a hash function.
+Equality of hash codes is a necessary, but not sufficient, condition
+for the original objects to be equal. 
+As integer comparisons are very efficient, performing an initial
+comparison of hash codes and then performing a detailed comparison
+only if the hash codes are equal can improve performance.
+The hash codes, in turn, can be mapped to a smaller set of integers,
+that can be used as an index, termed a hash index, to a rank one
+array, often termed a hash table or hash map.
+This document discusses the hash maps in the library.
```

Comment:
Here I think the text initially gives too much of an introduction to hash codes, outside of the context of hash maps. It might be better to start out explaining what a hash map is (in sufficiently vague terms). 

Not sure what the wording should be. Perhaps: "A hash map is a data-structure that maps keys to values. Such data structures occur in many languages, for example the python `dict`, the C++ `map`, ...".

Reply:
Leet me see what I can come up with.

Reply:
I have revised that paragraph.